src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -4222,3 +4236,366 @@ poolinfo_valid(PG_FUNCTION_ARGS)
+		}
+	}
+
+	return MetadataCache.tdigestAggTDigestPercentile3a;
+}
+
+
+/*
+ * TDigestExtensionAggTDigestPercentileOf2 performs a lookup for the Oid of the tdigest
+ * aggregate;
+ *   tdigest_percentile_of(tdigest, double precision)
+ *
+ * If the aggregate is not found InvalidOid is returned.
+ */
+Oid
+TDigestExtensionAggTDigestPercentileOf2()
+{
+	if (MetadataCache.tdigestAggTDigestPercentileOf2 == InvalidOid)
+	{
+		Oid tdigestSchemaOid = TDigestExtensionSchema();
+		if (tdigestSchemaOid != InvalidOid)
+		{
+			/* extension schema is found, lookup aggregate in schema */
+			char *schemaName = get_namespace_name(tdigestSchemaOid);
+			MetadataCache.tdigestAggTDigestPercentileOf2 = LookupFuncName(
+				list_make2(makeString(schemaName), makeString("tdigest_percentile_of")),
+				2, (Oid[]) { TDigestExtensionTypeOid(), FLOAT8OID }, true);
+		}
+	}
+
+	return MetadataCache.tdigestAggTDigestPercentileOf2;
+}
+
+
+/*
+ * TDigestExtensionAggTDigestPercentileOf2a performs a lookup for the Oid of the tdigest
+ * aggregate;
+ *   tdigest_percentile_of(tdigest, double precision[])
+ *
+ * If the aggregate is not found InvalidOid is returned.
+ */
+Oid
+TDigestExtensionAggTDigestPercentileOf2a(void)
+{
+	if (MetadataCache.tdigestAggTDigestPercentileOf2a == InvalidOid)
+	{
+		Oid tdigestSchemaOid = TDigestExtensionSchema();
+		if (tdigestSchemaOid != InvalidOid)
+		{
+			/* extension schema is found, lookup aggregate in schema */
+			char *schemaName = get_namespace_name(tdigestSchemaOid);
+			MetadataCache.tdigestAggTDigestPercentileOf2a = LookupFuncName(
+				list_make2(makeString(schemaName), makeString("tdigest_percentile_of")),
+				2, (Oid[]) { TDigestExtensionTypeOid(), FLOAT8ARRAYOID }, true);
+		}
+	}
+
+	return MetadataCache.tdigestAggTDigestPercentileOf2a;
+}
+
+
+/*
+ * TDigestExtensionAggTDigestPercentileOf3 performs a lookup for the Oid of the tdigest
+ * aggregate;
+ *   tdigest_percentile_of(double precision, int, double precision)
+ *
+ * If the aggregate is not found InvalidOid is returned.
+ */
+Oid
+TDigestExtensionAggTDigestPercentileOf3()
+{
+	if (MetadataCache.tdigestAggTDigestPercentileOf3 == InvalidOid)
+	{
+		Oid tdigestSchemaOid = TDigestExtensionSchema();
+		if (tdigestSchemaOid != InvalidOid)
+		{
+			/* extension schema is found, lookup aggregate in schema */
+			char *schemaName = get_namespace_name(tdigestSchemaOid);
+			MetadataCache.tdigestAggTDigestPercentileOf3 = LookupFuncName(
+				list_make2(makeString(schemaName), makeString("tdigest_percentile_of")),
+				3, (Oid[]) { FLOAT8OID, INT4OID, FLOAT8OID }, true);
+		}
+	}
+
+	return MetadataCache.tdigestAggTDigestPercentileOf3;
+}
+
+
+/*
+ * TDigestExtensionAggTDigestPercentileOf3a performs a lookup for the Oid of the tdigest
+ * aggregate;
+ *   tdigest_percentile_of(double precision, int, double precision[])
+ *
+ * If the aggregate is not found InvalidOid is returned.
+ */
+Oid
+TDigestExtensionAggTDigestPercentileOf3a(void)
```

Comment:
Feel all these functions could be made much shorter with a helper function

Reply:
Possibly. Now that we move away from the cached version I might make a helper function where we solely pass the signature of the function we are looking for. That should make these functions pretty compact aliases to the function signature.